src/transformers/models/pop2piano/feature_extraction_pop2piano.py

```diff
@@ -0,0 +1,288 @@
+        n_target_step = len(beatstep)
+        ext_beatstep = self.extrapolate_beat_times(beatstep, (n_bars + 1) * 4 + 1)
+
+        batch = []
+        for i in range(0, n_target_step, n_steps):
+            start_idx = i
+            end_idx = min(i + n_steps, n_target_step)
+
+            start_sample = int(ext_beatstep[start_idx] * self.sampling_rate)
+            end_sample = int(ext_beatstep[end_idx] * self.sampling_rate)
+            feature = audio[start_sample:end_sample]
+            batch.append(feature)
+        batch = pad_sequence(batch, batch_first=True, padding_value=padding_value)
+
+        return batch, ext_beatstep
+
+    def single_preprocess(
+        self,
+        beatstep,
+        audio=None,
+    ):
+        """Preprocessing method for a single sequence."""
+
+        if audio is not None:
+            if len(audio.shape) != 1:
+                raise ValueError(
+                    f"Expected `audio` to be a single channel audio input of shape to be (n, ) but found {audio.shape}."
+                )
+
+        if beatstep[0] > 0.01:
+            logger.warning(f"`beatstep[0]` is not 0. All beatstep will be shifted by {beatstep[0]}.")
+            beatstep = beatstep - beatstep[0]
+
+        batch, ext_beatstep = self.preprocess_mel(
+            audio,
+            beatstep,
+            n_bars=self.n_bars,
+            padding_value=self.padding_value,
+        )
+
+        return batch, ext_beatstep
+
+    def __call__(
+        self,
+        raw_audio: Union[np.ndarray, List[float], List[np.ndarray]],
+        sampling_rate: int,
+        steps_per_beat: int = 2,
+        return_tensors: Optional[Union[str, TensorType]] = "pt",
+        **kwargs,
+    ) -> BatchFeature:
+        """
+        Main method to featurize and prepare for the model one sequence. Please note that `Pop2PianoFeatureExtractor`
+        only accepts one raw_audio at a time.
+
+        Args:
+            raw_audio (`np.ndarray`, `List`):
+                Denotes the raw_audio.
+            sampling_rate (`int`):
+                Denotes the Sampling Rate of `raw_audio`.
+            steps_per_beat (`int`, *optional*, defaults to 2):
+                Denotes Steps per beat.
+            return_tensors (`str` or [`~utils.TensorType`], *optional*, defaults to 'pt'):
+                If set, will return tensors instead of list of python integers. Acceptable values are:
+                - `'pt'`: Return PyTorch `torch.Tensor` objects.
+                - `'np'`: Return Numpy `np.ndarray` objects.
+        """
+        is_batched = bool(
+            isinstance(raw_audio, (list, tuple))
+            and (isinstance(raw_audio[0], np.ndarray) or isinstance(raw_audio[0], (tuple, list)))
+        )
+        if is_batched:
+            raise ValueError(
+                "Pop2PianoFeatureExtractor only takes one raw_audio at a time, if you want to extract features from more than a single audio then you might need to call it multiple times."
```

Comment:
Maybe something like this:

```suggestion
                "Pop2PianoFeatureExtractor currently cannot handle batches. To extract features from more than one audio example, call the feature extractor multiple times."
```